### PR TITLE
Teach --cert to fetch from URLs

### LINKED ISSUE #281: Accept URLs in --cert
Certificates can be renewed periodically, and it's very important from a security POV that new sealed secrets are sealed with the latest certificate.

Forcing users to manually copy certificates locally is error-prone. Ideally the certificates are served in some trusted location, accessed via https (possibly live-served via an ingress from the controller itself).

Unfortunately `--cert` only accepts local files.

CC: @jjo

### PR DESCRIPTION
**Pull request by mkmik**
Closes #281

### CODE REVIEW

File: cmd/kubeseal/main.go [hunk 1]
```diff
@@ -142,15 +143,41 @@ func prettyEncoder(codecs runtimeserializer.CodecFactory, mediaType string, gv r
...
-	f, err := os.Open(certFile)
+func isFilename(name string) (bool, error) {
+	u, err := url.Parse(name)
 	if err != nil {
+		return false, err
+	}
+	return u.Scheme == "", nil
+}
+
+func openCertLocal(certURL string) (io.ReadCloser, error) {
```
Comment (jjo) [inline cmd/kubeseal/main.go]: IMO we should rename it to `openCertURI()`

File: cmd/kubeseal/main_test.go [hunk 2]
```diff
@@ -109,18 +109,23 @@ func TestParseKey(t *testing.T) {
...
 
-	f, err := openCert(certFile)
-	if err != nil {
-		t.Fatalf("Error reading test cert file: %v", err)
-	}
+	testCases := []string{"", "file://"}
+	for _, prefix := range testCases {
+		certURL := fmt.Sprintf("%s%s", prefix, certFile)
 
-	data, err := ioutil.ReadAll(f)
```
Comment (jjo) [inline cmd/kubeseal/main_test.go]: Would it make sense to also test `http` URI? (dunno how much much code-vs-lib test coverage we'd be increasing)

File: cmd/kubeseal/main.go (hunk 1 above)
Comment (mkmik) [reply cmd/kubeseal/main.go]: except a local filename is not an URI.

We even have a test that highlights that: we check for a file called `"/?this/file/cannot/possibly/exist/right?"`; if we treat arguments to `--cert` as URIs, possibly implicitly in the `file://` scheme, we'd require users to URI-encode the paths.

I thus opted for "Local" meaning "fetch the file from the POV of the client", as opposed from "use the k8s api server proxy to fetch things from within the cluster"

File: cmd/kubeseal/main_test.go (hunk 2 above)
Comment (mkmik) [reply cmd/kubeseal/main_test.go]: not sure; the only benefit I see would be to guard against us replacing the straight-forward `client.Get` call with something that treats `file://` specially and skips the `net/http` package. I don't find that possibility particularly likely, but you never know ... :-)

Comment (mkmik) [reply cmd/kubeseal/main_test.go]: anyway it turned out to be straightforward to test, so done

File: cmd/kubeseal/main.go (hunk 1 above)
Comment (mkmik) [reply cmd/kubeseal/main.go]: naming is hard; added comments

**Comment by mkmik** [comment]
PTAL

File: cmd/kubeseal/main.go (hunk 1 above)
Comment (jjo) [reply cmd/kubeseal/main.go]: > except a local filename is not an URI.

Looks like it is: https://tools.ietf.org/html/rfc8089

Comment (jjo) [reply cmd/kubeseal/main.go]: > naming is hard; added comments

Fair, comments "fulfill" it, but I'd still raise rfc8089 to claim the openCertURI() rename :)

Comment (mkmik) [reply cmd/kubeseal/main.go]: sure, rfc8089 is invoked when the user literally provides a parameter that starts with `file:...`

It's incorrect to assume that any host pathname that is valid to open with `os.Open` can be prefixed with `file:` and transformed to a rf8090 uri though.

I'll just make a couple of examples, using `curl` but the Go's FileTransport behaves in the same way.

1. 

```bash
$ echo foo >'/tmp/cert%E3%84%93.pem'
$ echo bar >'/tmp/certㄓ.pem'
$ cat '/tmp/cert%E3%84%93.pem'
foo
$ curl 'file:///tmp/cert%E3%84%93.pem'
**bar**
```

2.

```bash
$ echo foo >'/tmp/cert.pem'
$ echo bar >'/tmp/cert.pem?'
$ cat  '/tmp/cert.pem'
foo
$ curl 'file:///tmp/cert.pem'
foo
$ cat '/tmp/cert.pem?'
bar
$ curl 'file:///tmp/cert.pem?'
**foo**
```

I'll spare you the examples with windows backslashes.

Comment (mkmik) [reply cmd/kubeseal/main.go]: that said, I hear you; the naming is confusing. So I split it further and made it explicit what is a filename what is an URI

**Comment by Crevil** [comment]
This is super useful! Thanks for doing it. 🎉

**Pull request merged**